src/mpid/common/shm/mpidu_shm_alloc.c

```diff
@@ -148,7 +148,7 @@ static void *generate_random_addr(size_t size)
         iter--;
 
         if (iter == 0) {
-            map_pointer = -1ULL;
```

Comment:
I prefer `-1` or `UINTPTR_MAX` to this fix. I don't think there any need for the literal to be a ULL.

Reply:
I think `UINTPTR_MAX` will do here. `-1` always feels like a hack.